source/Symbol/SwiftASTContext.cpp

```diff
@@ -99,34 +99,32 @@
 #include "Plugins/SymbolFile/DWARF/DWARFASTParserSwift.h"
 
 #ifdef LLDB_CONFIGURATION_DEBUG
-#define VALID_OR_RETURN(value)                                                 \
+#define _VALID_OR_RETURN(_return)                                              \
   do {                                                                         \
-    lldbassert(!HasFatalErrors());                                             \
     if (HasFatalErrors()) {                                                    \
-      return (value);                                                          \
+      DiagnosticManager diagnostic_manager;                                    \
+      CopyDiagnostics(diagnostic_manager);                                    \
+      llvm::errs() << diagnostic_manager.GetString();                          \
+      lldbassert(!HasFatalErrors());                                           \
+      lldbassert(false);                                                       \
```

Comment:
It was just to get the right `assert` message (stringified `HasFatalError()`) and at the same time being guaranteed that we are going to die (hence the `lldbassert(false)`), but of course on non-debug builds we don't die, so `lldbassert` would end up printing 2x.

Retaining just the `lldbassert(false)`.